### ISSUE TITLE
Images emitted by file-loader doesn't show in assets.json

### ISSUE THREAD

**Issue by liximomo** [ztoben/assets-webpack-plugin#41]
``` js
loaders: [
  ...
  { 
    test: /\.(jpe?g|png|gif|svg)$/i, 
    loader: `file?name=[path][name].[ext]?[hash]&context=${context}!img`
  }
]
```

nothing show in assets.json except js and css

**Comment by dimchez** [comment]
@liximomo regexp that you are using does not match resulting file names for images that `file-loader` emits. As far as I can tell, names of images in assets array emitted by Webpack will look something like `some_path/some_image.svg?some_hash` which does not match `/\.(jpe?g|png|gif|svg)$/i` regexp.

Try using something like `/\.(jpe?g|png|gif|svg)\?./i`

**Comment by liximomo** [comment]
@dimchez  I don't think so. The regexp just tells webpack which file should use the loader.

**Comment by dimchez** [comment]
@liximomo Sorry, my bad. I was thinking about `assets-webpack-plugin` config, not `file-loader`. You need to set `assetsRegex` to override default regex, see [Options](https://github.com/kossnocorp/assets-webpack-plugin#options) for an example.

**Comment by liximomo** [comment]
@dimchez 

plugin config

``` js
new AssetsPlugin({
      filename: 'assetsMap.json',
      prettyPrint: true,
      assetsRegex: /\.(jpe?g|png|gif|svg)\?./i 
    }
```

webpack output

```

                                               Asset       Size  Chunks             Chunk Names
demo/img/header.png?61a77d7233b58c1aa55b7274fd3962aa    2.38 MB          [emitted]
                        test.js?e6e879a6c088dd635229  413 bytes       0  [emitted]  test
                        demo.js?e6e879a6c088dd635229  412 bytes       1  [emitted]  demo
           demo.css?cc5bc046f1fda29bc8ec4acccd61d9aa  320 bytes       1  [emitted]  demo
           test.css?7411b9041dad72eeed20b4bced355941   83 bytes       0  [emitted]  test
                    test.js.map?e6e879a6c088dd635229    4.25 kB       0  [emitted]  test
       test.css.map?7411b9041dad72eeed20b4bced355941  118 bytes       0  [emitted]  test
                    demo.js.map?e6e879a6c088dd635229    4.24 kB       1  [emitted]  demo
       demo.css.map?cc5bc046f1fda29bc8ec4acccd61d9aa  118 bytes       1  [emitted]  demo
```

assetsMap.json

``` js
{
  "test": {
    "js": "/assets/test.js?e6e879a6c088dd635229",
    "css": "/assets/test.css?7411b9041dad72eeed20b4bced355941"
  },
  "demo": {
    "js": "/assets/demo.js?e6e879a6c088dd635229",
    "css": "/assets/demo.css?cc5bc046f1fda29bc8ec4acccd61d9aa"
  }
}
```

No thing changed  :worried:

**Comment by b2whats** [comment]
https://github.com/kossnocorp/assets-webpack-plugin/issues/42

**Comment by dimchez** [comment]
@liximomo does it change anything if you remove `img-loader` from your loaders config, so change it to

``` js
{ 
  test: /\.(jpe?g|png|gif|svg)$/i, 
  loader: `file?name=[path][name].[ext]?[hash]&context=${context}`
}
```

**Comment by liximomo** [comment]
@dimchez  still nothing changed. 
Is it a file-loader problem or assets-webpack-plugin problem?

**Comment by dimchez** [comment]
@liximomo that's what I'm trying to figure out :) they way I usually approach such issues is simplifying configs and removing extra moving parts. E.g. in your case, that would be simplifying `file-loader` config to something like ``file?name=[path][name].[ext]` and removing `assetsRegex` option from `assets-webpack-plugin` config. Do you think you could try that?

**Comment by stoikerty** [comment]
On npm, for this package it says "_I'm looking for a new maintainer, please contact me if you are interested_". I assume you've taken ownership of the package @kossnocorp? Or are you looking for someone else to maintain it?

This version of `assets-webpack-plugin` needs to be released to npm. The plugin works if I use this git-repo directly, but you'll want to avoid using git as a dependency.

``` json
"devDependencies": {
  "assets-webpack-plugin": "git://github.com/kossnocorp/assets-webpack-plugin",
```
- _<small>[Getting Started with NPM (as a developer)](https://gist.github.com/coolaj86/1318304)</small>_
- _<small>[Publishing npm packages](https://docs.npmjs.com/getting-started/publishing-npm-packages)</small>_

**Comment by monteiro** [comment]
@stoikerty have you tried:

```
"devDependencies": {
    "assets-webpack-plugin": "^3.4.0",
```

It is what I have in my package.json

**Comment by neurosnap** [comment]
Any updates on this?  Is this effectively a dead repo now?

**Comment by kossnocorp** [comment]
@neurosnap no, I'm stuck in a big code refactoring which I started to learn the code base.

I am going to revert `master` to the state of 3.4.0 (no force pushes, no worries) and move development of 4.0 to `next` branch.

Sorry for the delay, everyone.

**Comment by psabharwal123** [comment]
any updates?

**Comment by vwong** [comment]
Why was this removed in 89d481933bdcbbcd3839396b4d93d0f579ad5672?

**Comment by stoikerty** [comment]
This might not work for everybody but I've moved to using a similar plugin called [manifest-revision-webpack-plugin](https://github.com/nickjj/manifest-revision-webpack-plugin) which helped me solve the issues I was facing with the assets-json-file I needed for static rendering with [dev-toolkit](https://github.com/stoikerty/dev-toolkit).

**Comment by vwong** [comment]
Thanks for the heads up. I'll look into that other plugin instead.

**Comment by Kronuz** [comment]
The problem is `file-loader` is not adding the emitted files to any chunk. It should be doing `chunk.files.push(filename)` somewhere....

**Comment by Kronuz** [comment]
Check this out: https://github.com/mariusbalcytis/extract-file-loader

**Comment by Kronuz** [comment]
I added a pull request: https://github.com/kossnocorp/assets-webpack-plugin/pull/65

**Comment by ztoben** [comment]
This should be fixed now in `3.7.0`.
